react/modules/extraEvents.ts

```diff
@@ -8,12 +8,14 @@ export async function sendExtraEvents(e: PixelMessage) {
         event: 'pageView',
         location: e.data.pageUrl,
         page: e.data.pageUrl.replace(e.origin, ''),
-        referrer: e.data.referrer,
+        referrer: e.data.referrer ? e.data.referrer : sessionStorage.getItem('gtmRefferer') ? sessionStorage.getItem('gtmRefferer') : '',
```

Comment:
What do you think about this suggestion?

```suggestion
        referrer: e.data.referrer ??  (sessionStorage.getItem('gtmRefferer') ?? ''),
```

Reply:
I also suggest you call the sessionStorage inside a try/catch block because if the user blocks the storage use, the page may break.